tests/unit/test_detect_target_contour.py

```diff
@@ -0,0 +1,286 @@
+) -> None:
+    """
+    Compare expected and actual detections.
+    """
+    assert len(actual.detections) == len(expected.detections)
+
+    # Using integer indexing for both lists
+    # pylint: disable-next=consider-using-enumerate
+    for i in range(0, len(expected.detections)):
+        expected_detection = expected.detections[i]
+        actual_detection = actual.detections[i]
+
+        assert expected_detection.label == actual_detection.label
+        np.testing.assert_almost_equal(
+            expected_detection.confidence,
+            actual_detection.confidence,
+            decimal=CONFIDENCE_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.x_1,
+            expected_detection.x_1,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.y_1,
+            expected_detection.y_1,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.x_2,
+            expected_detection.x_2,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.y_2,
+            expected_detection.y_2,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+
+def create_detections(detections_from_file: np.ndarray) -> detections_and_time.DetectionsAndTime:
+    """
+    Create DetectionsAndTime from expected.
+    Format: [confidence, label, x_1, y_1, x_2, y_2] .
+    """
+    assert detections_from_file.shape[1] == 6
+
+    result, detections = detections_and_time.DetectionsAndTime.create(0)
+    assert result
+    assert detections is not None
+
+    for i in range(0, detections_from_file.shape[0]):
+        result, detection = detections_and_time.Detection.create(
+            detections_from_file[i][2:],
+            int(detections_from_file[i][1]),
+            detections_from_file[i][0],
+        )
+        assert result
+        assert detection is not None
+        detections.append(detection)
+
+    return detections
+
+
+# ------------------------------------------------------------------------------------------------------------------
+@pytest.fixture()
+def detector() -> detect_target_contour.DetectTargetContour:  # type: ignore
+    """
+    Construct DetectTargetUltralytics.
+    """
+    detection = detect_target_contour.DetectTargetContour()
+    yield detection  # type: ignore
+
+
+# ---------------------------------------------------------------------------
+class TestDetector:
+    """
+    Tests `DetectTarget.run()` .
+    """
+
+    def test_multiple_landing_pads(
+        self,
+        detector: detect_target_contour.DetectTargetContour,
+    ) -> None:
+        """
+        Multiple images.
+        """
+
+        circle_data = [
+            [(200, 200), 400, False, [False, None, None]],
```

Comment:
This should be a type (e.g. objects of type `class FixtureProperties` go into some generator that outputs objects of type `class Fixture` which contain the numpy array and expected bounding boxes).